File: Programmers/hyeonyoung/멀리 뛰기.cpp
#include <string>
#include <vector>

using namespace std;

long long solution(int n)
{
    long long dp[2001];
    dp[0] = 1;
    dp[1] = 1;
    for (int i = 2; i <= n; ++i)
    {
        dp[i] = (dp[i - 1] + dp[i - 2]) % 1234567;
    }

    return dp[n];
}

File: Programmers/myeongju/멀리뛰기.java
import java.util.*;
import java.io.*;

class Solution {
    public long solution(int n) {
        
        long[] dp = new long[n+1];
        
        dp[1] = 1l;
        if(n>1) dp[2] = 2l;
        
        for(int i=3; i<=n; i++) 
            dp[i] = (dp[i-1] + dp[i-2]) % 1234567;
        
        return dp[n];
    }
}